subprojects/docs/src/docs/userguide/build_environment.adoc

```diff
@@ -156,32 +156,31 @@ You should check for existence of optional project properties before you access
-
 You can adjust JVM options for Gradle in the following ways:
 
-The `JAVA_OPTS` environment variable is used for the Gradle client, but not forked JVMs.
+The `org.gradle.jvmargs` Gradle property controls the VM running the build. It defaults to `-Xmx512m -XX:MaxMetaSpaceSize=256m`
 
-.Changing JVM settings for Gradle client JVM
+.Changing JVM settings for the build VM
 ----
-JAVA_OPTS="-Xmx2g -XX:MaxPermSize=256m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
+org.gradle.jvmargs=-Xmx2g -XX:MaxMetaSpaceSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
 ----
 
-You need to use the `org.gradle.jvmargs` Gradle property to configure JVM settings for the <<gradle_daemon.adoc#gradle_daemon, Gradle Daemon>>.
+The `JAVA_OPTS` environment variable controls the command line client, which is only used to display console output. It defaults to `-Xmx64m`
 
-.Changing JVM settings for forked Gradle JVMs
+.Changing JVM settings for the client VM
 ----
-org.gradle.jvmargs=-Xmx2g -XX:MaxPermSize=256m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+JAVA_OPTS="-Xmx64 -XX:MaxPermSize=64m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
 ----
 
 [NOTE]
 ====
-Many settings (like the Java version and maximum heap size) can only be specified when launching a new JVM for the build process. This means that Gradle must launch a separate JVM process to execute the build after parsing the various `gradle.properties` files.
-
-When running with the <<gradle_daemon.adoc#gradle_daemon, Gradle Daemon>>, a JVM with the correct parameters is started once and reused for each daemon build execution. When Gradle is executed without the daemon, then a new JVM must be launched for every build execution, unless the JVM launched by the Gradle start script happens to have the same parameters.
+There is one case where the client VM can also serve as the build VM:
+If you deactivate the <<gradle_daemon.adoc#gradle_daemon, Gradle Daemon>> and the client VM has the same settings as required for the build VM, the client VM will run the build directly.
+Otherwise the client VM will fork a new VM to run the actual build in order to honor the different settings.
 ====
 
-Certain tasks in Gradle also fork additional JVM processes, like the `test` task when using link:{javadocPath}/org/gradle/api/tasks/testing/Test.html#setMaxParallelForks-int-[Test.setMaxParallelForks(int)] for JUnit or TestNG tests. You must configure these through the tasks themselves.
+Certain tasks, like the `test` task, also fork additional JVM processes. You can configure these through the tasks themselves.
```

Comment:
Should we include a link to `JavaForkOptions` here?

Reply:
Yes, good idea.

Reply:
Actually, there are already samples for JavaCompile and Test here.

Reply:
Alright, I missed those. Nevermind then. 🙂